File: live-test/java/src/main/java/demo/StaticSuper.java
// Static-hiding fixture: StaticDerived.factory hides StaticBase.factory
// rather than overriding it, so the super probe reports no super method —
// hiding is not overriding.
package demo;

class StaticBase {
    static String factory() {
        return "base";
    }
}

class StaticDerived extends StaticBase {
    static String factory() {
        return "child";
    }
}

File: live-test/php/src/StaticSuper.php
<?php
// Static-method super fixture: StaticDerived::make() overrides the abstract
// static make() on StaticBase — static overrides super-resolve too.
namespace Demo;

abstract class StaticBase {
    abstract public static function make(): string;
}

class StaticDerived extends StaticBase {
    public static function make(): string {
        return "child";
    }
}
